src/v/cluster_link/link.cc

```diff
@@ -31,23 +32,74 @@ link::link(
 ss::future<> link::start() {
     vlog(
       cllog.info, "Starting cluster link {} ({})", _config.name, _config.uuid);
-    return ss::now();
+    for (auto& t : _tasks) {
+        if (should_start_task(t)) {
+            vlog(
+              cllog.info,
+              "Starting task {} for cluster link {} ({})",
+              t->name(),
+              _config.name,
+              _config.uuid);
+            auto res = co_await t->start();
```

Comment:
something to watch out for in stuff like this is the unexpected exception that terminates the loop. it can be built-in to `t->start()` or done at a higher level.